### DISCUSSION TITLE
Control  a Speaker

### DISCUSSION THREAD

**Discussion by tueddy** [HomeSpan/HomeSpan#265] [Q&A] [answered]
Hi,

just a beginner with HomeSpan and succeeded with first examples in few minutes, thank's for your great library!

Now i want to control an ESP32 based MP3-player. **NOT** streaming any audio, just control volume/mute and maybe later show battery status.
So i have created a new service of type speaker to handle volume and mute and added the missing category `Speakers=26` in Setting.h

Pairing is successful and i can see the speaker symbol as category. But in homeapp i see "This device is not supported in home-app".

Can you point me in the right direction? What type of service can be used here?

**Comment by HomeSpan** [comment, marked as answer]
Unfortunately the Home App does not support these functions.  I think they were deprecated a while back and Apple has not updated the HAP-R2 docs.  I have tried numerous ways of controlling a speaker/volume but have not yet succeeded.

**Comment by tueddy** [comment]
Hi, thank's for fast response! This is what i suspected of, speaker seems to be unsupported..

OK, main goal is to control play/pause/mute via Siri. This would be some normal switches e.g. lightbulb?
I have to evaluate if this makes sense at all for the project.

The Player has also a web-ui based on ESPAsyncWebserver. My tests showed that HomeSpan works fine together with ESPAsyncWebserver! I have initialized with `homeSpan.setPortNum(1201); `, so switch to another port. ESPAsyncWebserver runs on port 80. I got some warnings in serial output but all is working fine!

**Comment by HomeSpan** [comment]
You may want to consider using the Switch Service -- it contains just the On Characteristic.

And thanks for the info on ESPAsyncWebserver.   I had tried this before and it did not work (hence the FAQ where I recommend using just WebServer.h).  I had not revisited this since version 2 of the Arduino-ESP32 Board Manager was released, but Espressif made a lot of changes to the TCP libraries -- seems like they may have fixed this as well.

**Reply by tueddy** [reply]
I use PlatformIO and these (pinned) libraries:
`https://github.com/me-no-dev/ESPAsyncWebServer.git#1d46269`
`https://github.com/me-no-dev/AsyncTCP.git#ca8ac5f`

As said before all functionality is fine together with HomeSpan on port 1201. This are the serial output warnings:

`PietBox is READY!

Successfully connected to MyWLAN! IP Address: 192.168.178.56

Starting MDNS...

HostName:      HomeSpan-DF15B4557737.local:1201
Display Name:  PietBox
Model Name:    HomeSpan-ESP32
Setup ID:      HSPN

Starting HAP Server on port 1201 supporting 5 simultaneous HomeKit Controller Connections...

[E][WiFiClient.cpp:463] available(): fail on fd 0, errno: 9, "Bad file number"
[E][WiFiClient.cpp:463] available(): fail on fd 0, errno: 9, "Bad file number"
[E][WiFiClient.cpp:463] available(): fail on fd 0, errno: 9, "Bad file number"
[E][WiFiClient.cpp:463] available(): fail on fd 0, errno: 9, "Bad file number"
[E][WiFiClient.cpp:463] available(): fail on fd 0, errno: 9, "Bad file number"
Aktuelle Batteriespannung: 4.16 V
RSSI: -78 dBm`

**Comment by danilkorotkov** [comment]
Async has many useful things, ex weblog of serial output
